### PR TITLE
A bunch of fixes for Inductor + dynamic shapes enablement

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92609
* #92597
* #92397

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -3681,11 +3681,30 @@ def op_mul(a, b):
 def op_add(a, b):
     return a + b
 
+@register_lowering(operator.sub)
+def op_add(a, b):
+    return a - b
```
Comment (ezyang) [inline torch/_inductor/lowering.py:3686]: Fix the name plz

**Review by ezyang — APPROVED** [review]

File: torch/fx/experimental/proxy_tensor.py [hunk 2]
```diff
@@ -392,7 +392,7 @@ def can_handle_tensor(x):
 
 class PythonKeyTracer(Tracer):
     def __init__(self):
-        super().__init__()
+        super().__init__(autowrap_modules=())
```
Comment (Chillee) [inline torch/fx/experimental/proxy_tensor.py:395]: Oh, this is a little bit worrisome... I'm not quite sure how to handle this. The issue is that FX autowraps `math.sqrt` (and other such functions) into a special lambda. Then, we have no idea how to lower it.

Comment (voznesenskym) [reply torch/fx/experimental/proxy_tensor.py:395]: Why does it wrap it into a special lambda? Can we have fx have its own impl that we redirect through that isn't a lambda? fx.math_sqrt or smth?

Comment (Chillee) [reply torch/fx/experimental/proxy_tensor.py:395]: yeah I think that would be reasonable/preferable.

Comment (Chillee) [reply torch/fx/experimental/proxy_tensor.py:395]: Actually, I'm not sure this is worrisome 🤔 It doesn't seem to be breaking anything.

Comment (Chillee) [reply torch/fx/experimental/proxy_tensor.py:395]: So I'm gonna land it if it passes tests :^)

**Pull request closed without merging**